scripts/benchmark_multithreaded.py

```diff
@@ -0,0 +1,190 @@
+    parser.add_argument('-q', '--queries', action='store', type=int, metavar='Q', nargs='+', help='Specify the TPC-H queries that will be benchmarked')
+    parser.add_argument('--chunk-size', action='store', type=int, metavar='S', help='Specify maximum chunk size (default: Maximum available)')
+    parser.add_argument('--result-dir', action='store', type=str, metavar='DIR', default='results', help='Directory where the result folder will be stored (default: \'results/\')')
+    parser.add_argument('--result-name', action='store', type=str, metavar='NAME', help='Directory where the actual results will be stored (default: current datetime)')
+    parser.add_argument('--build-dir', action='store', type=str, metavar='DIR', required=True, help='Directory that contains the hyriseBenchmarkTPCH executable')
+
+    core_group = parser.add_mutually_exclusive_group()
+    core_group.add_argument('-c', '--cores', action='store', type=int, metavar='C', nargs='+', help='List of cores to be used for the benchmarks')
+    core_group.add_argument('--max-cores', action='store', type=int, metavar='C', help='Number of cores this machine has available')
+
+    run_group = parser.add_mutually_exclusive_group(required=True)
+    run_group.add_argument('--fixed-runs', action='store', type=int, metavar='N', help='Fixed number of runs each query is executed')
+    run_group.add_argument('--runs-per-core', action='store', type=int, metavar='N', help='Number of runs per core each query is executed')
+    
+    return parser.parse_args()
+
+def verbose_print(verbose, message):
+    if verbose: print('python> ' + message)
+
+def get_core_counts(args):
+    if args.cores:
+        return args.cores
+    max_core_count = args.max_cores if args.max_cores else MAX_CORE_COUNT
+    if max_core_count < 8:
+        step = -1
+    elif max_core_count < 16:
+        step = -int(max_core_count / 4)
+    else:
+        step = -int(max_core_count / 8)
+    core_counts = list(range(max_core_count, 0, step))
+    if core_counts[-1] != 1:
+        core_counts.append(1)
+    core_counts.append(0)
+    return core_counts
+
+def get_formatted_queries(args):
+    queries = args.queries if args.queries else DEFAULT_TPCH_QUERIES
+    formatted_queries = []
+    for query in queries:
+        formatted_queries += ['-q', str(query)]
+    return formatted_queries
+
+def is_square(n):
+    return n ** 0.5 == int(n ** 0.5)
+
+def get_subplot_row_and_column_count(num_plots):
+    while not is_square(num_plots):
+        num_plots += 1
+    return num_plots ** 0.5
+
+def run_benchmarks(args, core_counts, executable, result_dir):
+    benchmark_run = 0
+    for core_count in core_counts:
+        benchmark_run += 1
+        verbose_print(args.verbose, 'Starting benchmark run ' + str(benchmark_run) + ' of ' + str(len(core_counts)) + ' ...')
+
+        if core_count == 0:
+            use_scheduler = 'false'
+            number_of_runs = args.runs_per_core if args.runs_per_core else args.fixed_runs
```

Comment:
Thinking about this again, I still think this is more complicated than it has to be. What if we allow `--runs=0` on the Hyrise side which means "run until time is up". Wouldn't that solve the problem here?

Reply:
Not sure what problem you mean here. If I specify `--runs=0` (and if that's allowed on the Hyrise side), then that will be used and passed to Hyrise. But if I specify `--runs-per-core=X`, then that instead will be used to determine the number of max query runs for this benchmark run.